static/app/components/draggableTabs/index.tsx

```diff
@@ -75,6 +78,35 @@ export function Tabs<T extends string | number>({
   );
 }
 
+export interface Tab {
+  content: React.ReactNode;
+  key: Key;
+  label: string;
+}
+
+export interface DragAndDropTabBarProps {
+  tabs: Tab[];
+}
+
+export function DraggableTabBar(props: DragAndDropTabBarProps) {
+  const [tabs, setTabs] = useState<Tab[]>(props.tabs);
```

Comment:
might need a useEffect to update this state if the props.tabs change otherwise it will only use the props.tabs from the first render